src/common/packet/game/AvatarFirstTimeEventMessage.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "common/packet/opcodes.h"
+#include "common/bitstream.h"
+
+/**
+    Dispatched to the server when the player encounters something for the very first time in their campaign.
+    For example, the first time the player rubs up against a game object with a yellow exclamation point.
+    For example, the first time the player draws a specific weapon.
+
+    When the first time events (FTE's) are received, battle experience is awarded.
+    Text information about the object will be displayed.
+    A certain itemized checkbox under the 'Training' tab that corresponds is marked off.
+    This indicates which tracked items have not yet been "encountered."
+    These effects only happen once per character/campaign.
+    (The Motion Sensor will occasionally erronously display the information window on repeat encounters.
+    No additional experience, though.)
+
+    First time events (FTE's) are recorded in a great list in the middle of player ObjectCreateMessage data.
+    Tutorial complete events are enqueued nearby.
+    @property player_uid the player
+    @property object_id the game object that triggers the event
+    @property unk na
+    @property event_name the string name of the event
+*/
+class AvatarFirstTimeEventMessage {
+public:
+    uint32_t unk;
+    uint16_t avatar_uid;
+    uint16_t object_id;
+    std::string event_name;
```

Comment:
Can you move the unk var here to below object_id to match the order of vars in the packet data (i.e. decode func below)? Otherwise I could see glancing that this and thinking that this is the order of the data in the packet, when it's actually different in the decode func.